pkg/master/master.go

```diff
@@ -781,11 +782,14 @@ func (m *Master) api_v1() *apiserver.APIGroupVersion {
 func (m *Master) expapi(c *Config) *apiserver.APIGroupVersion {
 	controllerStorage := expcontrolleretcd.NewStorage(c.DatabaseStorage)
 	autoscalerStorage := horizontalpodautoscaleretcd.NewREST(c.DatabaseStorage)
+	nodeMetricsStorage := nodemetrics.NewREST(c.DatabaseStorage)
 
 	storage := map[string]rest.Storage{
 		strings.ToLower("replicationControllers"):       controllerStorage.ReplicationController,
 		strings.ToLower("replicationControllers/scale"): controllerStorage.Scale,
 		strings.ToLower("horizontalpodautoscalers"):     autoscalerStorage,
+		// /api/experimental/nodeMetrics/{myNode}
```

Comment:
We enforce it in api_installer - the current code pattern is to set your
storage names like they are in master.go (calling lower on the camelCased
name).  Someone who wanted to change the pattern should remove all the
strings.ToLower and switch them all to lower case.  But for now be
consistent with the code you are changing.

On Fri, Aug 21, 2015 at 4:52 PM, Daniel Martí <notifications@github.com>
wrote:

> In pkg/master/master.go
> <https://github.com/kubernetes/kubernetes/pull/12952#discussion_r37677052>
> :
>
> >
> >  	storage := map[string]rest.Storage{
> >  		strings.ToLower("replicationControllers"):       controllerStorage.ReplicationController,
> >  		strings.ToLower("replicationControllers/scale"): controllerStorage.Scale,
> >  		strings.ToLower("horizontalpodautoscalers"):     autoscalerStorage,
> > +		// /api/experimental/nodeMetrics/{myNode}
>
> Since github doesn't show that comment properly since it's hidden, I'll
> just copy the comment.
>
> @nikhiljindal <https://github.com/nikhiljindal> said "Not sure if we
> enforce it, but its lowercase for all our existing resources."
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12952/files#r37677052>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift